lib/assert.js

```diff
@@ -300,13 +300,15 @@ function setEquiv(a, b, strict, actualVisitedObjects) {
 
   outer: for (const val1 of a) {
     if (!b.has(val1)) {
-      // The value doesn't exist in the second set by reference, so we'll go
-      // hunting for something thats deep-equal to it. Note that this is O(n^2)
-      // complexity, and will get slower if large, very similar sets / maps are
-      // nested inside. Unfortunately there's no real way around this.
-      for (const val2 of b) {
-        if (_deepEqual(val1, val2, strict, actualVisitedObjects)) {
-          continue outer;
+      if (!strict || typeof val1 === 'object') {
```

Comment:
We don't need to search for functions because the only way deepEqual will consider functions equal to one another is by reference equality. The null check makes some sense, although I think it'll make a very small difference in practice because putting null in a set and using null as a key are rare. (In my code, null popping up in either of those places is usually an indication of a bug)